assets/hello_world.c

```diff
@@ -0,0 +1,7 @@
+// This is a file use to compile some of the binaries in this
+// directory for testing purposes.
+#include <stdio.h>
+
+int main(int argc, char *argv[]) {
```

Comment:
I'm not sure this is the best example file for .so + .a files;

I'd expect something more like:

```c
extern int foo(int x) {
  return x * 2;
}
```

Since it's meant to have some public symbol that other downstream binaries/libraries link against and use.

MachO does allow multi-arch executables (i.e., something with a main and the executable flags set in macho headers, etc.), but I'd rather test the .so + .a versions, so executables are just .so's with special header values.